ndscan/experiment/parameters.py

```diff
@@ -213,18 +215,23 @@ class ParamHandle:
     :param owner: The owning fragment.
     :param name: The name of the attribute in the owning fragment bound to this
         object.
+    :param parameter: The parameter associated with this handle.
     """
-    def __init__(self, owner: Any, name: str):
-        # `owner` will typically be a Fragment instance; no type hint to avoid circular
-        # dependency.
+    def __init__(self, owner: "Fragment", name: str, parameter):
         self.owner = owner
         self.name = name
+        self._parameter = parameter
+
         assert name.isidentifier(), ("ParamHandle name should be the identifier it is "
                                      "referred to as in the owning fragment.")
 
         self._store = None
         self._changed_after_use = True
 
+    def get_parameter(self) -> "ParamBase":
+        """Returns the parameter associated with this handle."""
+        return self._parameter
+
```

Comment:
Can just be a public field like `owner`/`name`. Python supports properties anyway if this ever needs to change (or we really wanted to enforce read-only-ness).

Reply:
Happy to do that if you prefer